src/libarena/lib.rs

```diff
@@ -337,38 +329,31 @@ impl<'longer_than_self> Arena<'longer_than_self> {
             }
         }
     }
-}
 
-#[test]
-fn test_arena_destructors() {
-    let arena = Arena::new();
-    for i in 0..10 {
-        // Arena allocate something with drop glue to make sure it
-        // doesn't leak.
-        arena.alloc(|| Rc::new(i));
-        // Allocate something with funny size and alignment, to keep
-        // things interesting.
-        arena.alloc(|| [0u8, 1u8, 2u8]);
+    /// Allocates a slice of bytes of requested length. The bytes are not guaranteed to be zero
+    /// if the arena has previously been cleared.
+    pub fn alloc_bytes(&self, len: usize) -> &mut [u8] {
+        unsafe {
+            let ptr = self.alloc_copy_inner(len, 1);
```

Comment:
It's terrifying but accurate to think of `+` with a user-suppiled integer as an "arbitrary value attack".